[brianchanshiyuan] iP

brianchanshiyuan opened the pull request.

brianchanshiyuan committed:
Delete Duke

brianchanshiyuan committed:
Level 0. Rename, Greet, Exit

brianchanshiyuan committed:
Level 1. Echo

brianchanshiyuan committed:
Level 2. Add, List

brianchanshiyuan committed:
Level 3. Mark as Done

brianchanshiyuan committed:
Level 4. ToDos, Events, Deadlines

brianchanshiyuan committed:
Refactor input processing and improve naming conventions

Extracted handling of ToDos, Events, and Deadlines into separate methods to simplify processInput. This improves readability and maintainability.

Replaced arbitrary constants with appropriately named capital-letter constants. Refined variable names for better clarity and consistency.

brianchanshiyuan committed:
Refactor input processing and improve naming conventions

Extracted handling of ToDos, Events, and Deadlines into separate methods to simplify processInput. This improves readability and maintainability.

Replaced arbitrary constants with appropriately named capital-letter constants. Refined variable names for better clarity and consistency.

brianchanshiyuan committed:
Refactor input processing and improve naming conventions

Extracted handling of ToDos, Events, and Deadlines into separate methods to simplify processInput. This improves readability and maintainability.

Replaced arbitrary constants with appropriately named capital-letter constants. Refined variable names for better clarity and consistency.

brianchanshiyuan committed:
Revert "Refactor input processing and improve naming conventions"

This reverts commit 38281eb1fdf9907ad1bee2a7aeb2625dec39bbbc.

brianchanshiyuan committed:
Reapply "Refactor input processing and improve naming conventions"

This reverts commit 0567cb1755b3548ffa417f600cee800f0ebd51a9.

brianchanshiyuan committed:
Add expected output file for task list testing

brianchanshiyuan committed:
Removed unnecessary lines and spacings

brianchanshiyuan committed:
Add runtest.bat and input.txt for automated output generation and comparison

shuu4 commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,191 @@
+}
+
+class Event extends Task {
+    protected String from;
```
Good naming convention! All classes are nouns.

shuu4 commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,191 @@
+                System.out.println("Invalid format! Use: event [task] /from [start] /to [end]");
+            }
+        }
+    }
```
Can try to reduce indentation and avoid arrowhead nesting to improve readability

shuu4 commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,191 @@
+            break;
+        case "event":
+            processEvent(inputParts);
+            break;
```
I like your use of switch here instead of multiple if-else blocks. The logic is well-organised!

shuu4 commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,191 @@
+        System.out.println(SEPARATOR);
+    }
+
+    private static void markTask(String input, boolean isDone) {
```
Good boolean name that starts with 'is'

brianchanshiyuan committed:
Refactor main function to remove arrowhead structure

Extracted logic into separate methods to improve readability and maintainability.
Main function now delegates responsibilities to helper methods, reducing nesting.
Ensured the program flow remains unchanged while enhancing code clarity.

brianchanshiyuan committed:
Edited the error messages to make them more specific.

brianchanshiyuan committed:
Add delete command using fixed-size array

brianchanshiyuan committed:
Migrate task storage to ArrayList

Replaces the fixed-size array with an ArrayList to allow for dynamic task list resizing. This eliminates the hard limit on the number of tasks.

brianchanshiyuan committed:
Implement task persistence using file storage

The task list is now persisted to disk using a file (./data/Nova.txt). This allows tasks to be saved between sessions, so the user's task list is preserved when the program is closed and reopened.  Handles file creation and directory creation if they don't exist.

Merged branch-Level-7 and branch-Level-6

# Conflicts:
#	src/main/java/Nova.java

brianchanshiyuan committed:
Implement task persistence using file storage

The task list is now persisted to disk using a file (./data/Nova.txt). This allows tasks to be saved between sessions, so the user's task list is preserved when the program is closed and reopened.  Handles file creation and directory creation if they don't exist.

brianchanshiyuan committed:
Revert "Implement task persistence using file storage"

This reverts commit 3d85a8376dce6fe68f143ed4b8b78f1b759a60d3.

brianchanshiyuan committed:
Revert "Implement task persistence using file storage"

This reverts commit 3d85a8376dce6fe68f143ed4b8b78f1b759a60d3.

Alfred-Goh02 commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,225 @@
+        processUserInput();
+        printExitMessage();
+    }
+
```
Consider separating printing methods into its own class

Alfred-Goh02 commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,225 @@
+            //System.out.println("Invalid command! Use: todo, deadline, event, mark, unmark, list, or bye.");
+        }
+    }
+
```
You can separate different commands into a different class as well.

Alfred-Goh02 commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,225 @@
+public class Nova {
+    public static final String SEPARATOR = "____________________________________________________________";
+    private static final int MAX_TASKS = 100;
+    private static final Task[] tasks = new Task[MAX_TASKS];
```
Consider using array list which is easily mutable.

Alfred-Goh02 commented on src/main/java/Nova.java:
```diff
@@ -0,0 +1,225 @@
+    public String toString() {
+        return "[E]" + super.toString() + " (" + from + " " + to + ")";
+    }
+}
```
Good job over all, consider separating your classes into different files as well, this will be important when working with multiple people as you will have less occurrence of merge conflicts.

brianchanshiyuan committed:
Implement task persistence using file storage

The task list is now persisted to disk using a file (./data/nova.txt). This allows tasks to be saved between sessions, so the user's task list is preserved when the program is closed and reopened.  Handles file creation and directory creation if they don't exist.

brianchanshiyuan committed:
Implement task persistence using file storage

The task list is now persisted to disk using a file (./data/nova.txt). This allows tasks to be saved between sessions, so the user's task list is preserved when the program is closed and reopened.  Handles file creation and directory creation if they don't exist.

brianchanshiyuan committed:
Added nova.txt. Removed duke.txt

brianchanshiyuan committed:
Merge branch 'branch-Level-6'

brianchanshiyuan committed:
Added nova.txt. Removed duke.txt

brianchanshiyuan committed:
Revert "Added nova.txt. Removed duke.txt"

This reverts commit edcb65f8089de0b1d773d1f34a717b7d9873bb55, reversing
changes made to 5329a1f0a52c3531f7832b6eb85d08fe4485d435.

brianchanshiyuan committed:
Merged branch-Level-6 and branch-Level-7.

Implemented save and delete function

brianchanshiyuan committed:
Nova.txt

brianchanshiyuan committed:
remerge with github repository

brianchanshiyuan committed:
Implemented Jar Files

brianchanshiyuan committed:
Added JavaDoc

brianchanshiyuan committed:
Introduced more OOP

brianchanshiyuan committed:
Added find function

brianchanshiyuan committed:
Merge branch 'master' into branch-Level-9

brianchanshiyuan committed:
Added JavaDoc

brianchanshiyuan committed:
Merge remote-tracking branch 'origin/master'

brianchanshiyuan committed:
Updated and Improved README.md

brianchanshiyuan committed:
Updated and Improved README.md

brianchanshiyuan committed:
Updated and Improved README.md

brianchanshiyuan committed:
Merge branch 'master' into branch-A-JavaDoc

brianchanshiyuan committed:
Merge pull request #1 from brianchanshiyuan/branch-Level-9

Level-9

brianchanshiyuan committed:
Merge branch 'master' into branch-A-JavaDoc

brianchanshiyuan committed:
Merge pull request #2 from brianchanshiyuan/branch-A-JavaDoc

Merge A-JavaDoc with master

brianchanshiyuan committed:
Updated and added more comments

brianchanshiyuan committed:
fixed typo error from "string[]" to "String[]"

brianchanshiyuan committed:
fixed typo error from "string[]" to "String[]"

brianchanshiyuan committed:
fixed typo error from "string[]" to "String[]"

brianchanshiyuan committed:
Added user guide

brianchanshiyuan committed:
Added user guide

brianchanshiyuan committed:
Merge remote-tracking branch 'origin/master'

brianchanshiyuan committed:
Added user guide

brianchanshiyuan committed:
edited .gitignore
refactored code into new classes

brianchanshiyuan committed:
Merge branch 'master' of https://github.com/brianchanshiyuan/ip

# Conflicts:
#	data/nova.txt
